src/main/java/org/spongepowered/api/map/decoration/MapDecoration.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.decoration;
+
+import com.flowpowered.math.vector.Vector2i;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.map.MapCanvas;
+import org.spongepowered.api.map.MapInfo;
+import org.spongepowered.api.util.Direction;
+
+/**
+ * A {@code MapDecoration} represents a symbol that may be placed at a specific
+ * point on a {@link MapInfo map}, which exists as a separate layer on top of a
+ * {@link MapCanvas}. A common example of a Decoration is the player marker.
+ *
+ * <p>Unlike a {@link MapCanvas} which is a relatively static feature of a
+ * {@link MapInfo}, the position and rotation of decorations can be updated
+ * more easily.</p>
+ *
+ * <p>The co-ordinate system used when getting or setting a decoration's
+ * location on a map is independent of any world a map may be based on.
+ * Instead, valid co-ordinates range from {@link Byte#MIN_VALUE -128} to
+ * {@link Byte#MAX_VALUE 127}. The bottom left corner of the map is given by the
+ * co-ordinates (-128, -128).</p>
+ */
+public interface MapDecoration extends DataSerializable {
+
+    /**
+     * Creates a {@link Builder} for generating {@link MapDecoration}s.
+     *
+     * @return A {@link Builder}
+     */
+    static Builder builder() {
```

Comment:
It might be useful to have 

```
MapDecoration of(final MapDecorationType type) {
    return MapDecoration.builder().type(type).build();
}
```

as you can set the position and rotation outside of the builder.